packages/lib/tagOnResponse/service.ts

```diff
@@ -3,9 +3,27 @@ import "server-only";
 import { prisma } from "@formbricks/database";
 import { TTagsCount, TTagsOnResponses } from "@formbricks/types/tags";
 import { responseCache } from "../response/cache";
+import { SERVICES_REVALIDATION_INTERVAL } from "../constants";
+import { unstable_cache } from "next/cache";
+import { tagOnResponseCache } from "./cache";
+import { validateInputs } from "../utils/validate";
+import { ZId } from "@formbricks/types/environment";
 
-export const getTagOnResponseCacheTag = (tagId: string, responseId: string) =>
-  `tagsOnResponse-${tagId}-${responseId}`;
+const selectTagsOnResponse = {
+  response: {
```

Comment:
If we go over the tag, we get to the environmentId faster with much less database calls 😊
```ts
{
  tag: {
    select: {
      environmentId,
    },
  },
};
```